### PR TITLE
Update sass-mq tests to cover more scenarios

### PR DESCRIPTION
**Pull request by domoscargin**
We're ultimately replacing this mixin with our own code, so we want to make sure we're matching what it does exactly.

This adds a few more cases to tests for edge cases so we know how to deal with them.

Part of #6235

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/helpers/media-queries.unit.test.js [hunk 1]
```diff
@@ -100,27 +116,34 @@ describe('@mixin govuk-media-query', () => {
     })
   })
 
-  it('allows you to target combined min-width and max-width using numeric values', async () => {
-    const sass = `
+  it.each([
+    ['20em', '40em', '20em', '40em'],
+    ['20px', '40px', '1.25em', '2.5em'],
+    ['20', '40px', '1.25em', '2.5em']
```
Comment (querkmachine) [inline packages/govuk-frontend/src/govuk/helpers/media-queries.unit.test.js]: Based on the prior tests it seems like these are meant to be unquoted numbers. Is that the case? 

```suggestion
    [20, 40, '1.25em', '2.5em']
```

**Review by querkmachine — APPROVED** [review]
One question, otherwise looks good to me!

**Review by querkmachine — APPROVED** [review]
One question, otherwise looks good to me!

File: packages/govuk-frontend/src/govuk/helpers/media-queries.unit.test.js (hunk 1 above)
Comment (domoscargin) [reply packages/govuk-frontend/src/govuk/helpers/media-queries.unit.test.js]: Ah yeah, was trying to test mixed inputs. I'll use an unquoted number and the px unit.
